src/pulsar_producer.erl

```diff
@@ -427,12 +425,62 @@ handle_response({connected, _ConnectedData}, State = #{
         sock := Sock,
         opts := Opts,
         request_id := RequestId,
-        producer_id := ProducerId,
-        partitiontopic := Topic
+        partitiontopic := PartitionTopic
     }) ->
     start_keepalive(),
-    pulsar_socket:send_create_producer_packet(Sock, Topic, RequestId, ProducerId, Opts),
+    %% if Pulsar went down and then restarted later, we must issue a
+    %% LookupTopic command again after reconnecting.
+    %% https://pulsar.apache.org/docs/2.10.x/developing-binary-protocol/#topic-lookup
+    %% > Topic lookup needs to be performed each time a client needs
+    %% > to create or reconnect a producer or a consumer. Lookup is used
+    %% > to discover which particular broker is serving the topic we are
+    %% > about to use.
+    %% Simply looking up the topic (even from a distinct connection)
+    %% will "unblock" the topic so we may send messages to it.  The
+    %% producer may be started only after that.
+    pulsar_socket:send_lookup_topic_packet(Sock, PartitionTopic, RequestId, Opts),
     {keep_state, next_request_id(State)};
+handle_response({lookupTopicResponse, #{error := Reason, message := Msg, response := 'Failed'}},
+                State) ->
+    log_error("error looking up topic: ~p", #{msg => Msg, error => Reason}, State),
+    try_close_socket(State),
+    ?NEXT_STATE_IDLE_RECONNECT(State);
+handle_response({lookupTopicResponse, #{} = Response}, State0) ->
+    #{ opts := Opts
+     , partitiontopic := PartitionTopic
+     , broker_server := BrokerServer0
+     , sock := Sock
+     , producer_id := ProducerId
+     , request_id := RequestId
+     } = State0,
+    %% TODO: unify this with similar handler from `pulsar_client' ???
+    ServiceURL =
```

Comment:
Ideally, it looks like it would be better to call the `pulsar_client` which originated this producer in the first place, to avoid this duplication.  

But we currently don't have any references (pid nor clientid) in the producer state that we may use to call the `pulsar_client`...  That'll be probably a bigger change and be more tricky to do with relup.